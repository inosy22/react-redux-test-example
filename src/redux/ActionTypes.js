const ActionTypes = {
  ADD_TODO: 'ADD_TODO',
  FILTER_COMPLETED_TODO: 'FILTER_COMPLETED_TODO',
}
export default ActionTypes
