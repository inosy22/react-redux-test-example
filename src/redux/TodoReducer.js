import ActionTypes from './ActionTypes'

const TodoReducer = (state, action) => {
  switch (action.type) {
    case ActionTypes.ADD_TODO: 
      return Object.assign({}, state, {
        todos: [
          ...state.todos, 
          {
            id: action.payload.id,
            text: action.payload.text,
            completed: action.payload.completed,
          }
        ]
      })
    case ActionTypes.FILTER_COMPLETED_TODO:
      return Object.assign({}, state, {
        showCompleted: action.payload.showCompleted
      })
    default:
      return state
  }
}
export default TodoReducer