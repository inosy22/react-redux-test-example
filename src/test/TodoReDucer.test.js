import snapshotDiff from 'snapshot-diff'
import TodoReducer from '../redux/TodoReducer'
import ActionTypes from '../redux/ActionTypes'

const defaultState = {
  showCompleted: false,
  todos: [],
}

describe('TodoReducerTest', () => {
  test('', () => {
    const action = {
      type: ActionTypes.ADD_TODO,
      payload: {
        id: 1,
        text: 'test',
        completed: false,
      }
    }
    expect(
      snapshotDiff(defaultState, TodoReducer(defaultState, action))
    ).toMatchSnapshot()
  })
})