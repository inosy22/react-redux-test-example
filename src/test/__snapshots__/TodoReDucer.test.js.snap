// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoReducerTest  1`] = `
"Snapshot Diff:
- First value
+ Second value

  Object {
    \\"showCompleted\\": false,
-   \\"todos\\": Array [],
+   \\"todos\\": Array [
+     Object {
+       \\"completed\\": false,
+       \\"id\\": 1,
+       \\"text\\": \\"test\\",
+     },
+   ],
  }"
`;
